docs/developer/contributing/running-kibana-with-apm.asciidoc

```diff
@@ -0,0 +1,44 @@
+[[running-kibana-with-apm]]
+== Running Kibana with the APM Agent Locally
+
+With an application as varied and complex as Kibana has become, it's not practical or scalable to craft all possible performance measurements by hand ahead of time. As such, we need to rely on tooling to help us catch things we may otherwise have missed.
+
+For example, say you implement a brand new feature, plugin or service but don't quite know how it will impact Kibana's performance as a whole. APM allows us to not only spot that something is slow, but also hints at why it might be performing slowly (for example, if a function is slow on specific types of inputs).
+
+The net of metrics captured by APM are both a wide and deep because the entire application is instrumented at runtime and we simply take a sample of these metrics. This means that we don't have to know what we need to measure ahead of time, we'll instead just get (most) of the data we're likely going to need by default.
+
+This type of data can help us identify unknown bottlenecks, spot when a performance regression may have been introduced, and inform how the performance of Kibana is changing between releases. Using APM allows us to be proactive in getting ahead of potential performance regressions before they are released.
+
+The easiest and recommended way of running Kibana with the APM agent locally is to use the solution provided by the https://github.com/elastic/apm-integration-testing[apm-integration-testing] repo. You’ll need https://www.docker.com/community-edition[Docker] and https://docs.docker.com/compose/install/[Docker Compose] to use the tool.
+
+[discrete]
+=== Quick start guide
+
+1. Clone https://github.com/elastic/apm-integration-testing[elastic/apm-integration-testing]
+2. Change into the apm-integration-testing repo:
+** `cd apm-integration-testing`
+3. Run {es} and the APM servers without Kibana using:
```

Comment:
```suggestion
3. Run {es} and the APM servers without running Kibana:
```

Reply:
@vigneshshanmugam Moving it makes sense, thanks for reminding us about that doc!